baybe/surrogates/gaussian_process/core.py

```diff
@@ -113,11 +114,74 @@ class GaussianProcessSurrogate(Surrogate):
     _model = field(init=False, default=None, eq=False)
     """The actual model."""
 
+    # Transfer learning fields
+    _prior_gp = field(init=False, default=None, eq=False)
+    """Prior GP to extract mean/covariance from for transfer learning."""
+
+    _transfer_mode: Literal["mean", "kernel"] | None = field(
+        init=False, default=None, eq=False
+    )
+    """Transfer learning mode: 'mean' uses prior as mean function, 'kernel' uses prior covariance."""
+
     @staticmethod
     def from_preset(preset: GaussianProcessPreset) -> GaussianProcessSurrogate:
         """Create a Gaussian process surrogate from one of the defined presets."""
         return make_gp_from_preset(preset)
 
+    @classmethod
+    def from_prior_gp(
+        cls,
+        prior_gp,
+        transfer_mode: Literal["mean", "kernel"] = "mean",
+        kernel_factory: KernelFactory | None = None,
+        **kwargs,
+    ) -> GaussianProcessSurrogate:
+        """Create a GP surrogate from a prior GP.
+
+        Args:
+            prior_gp: Fitted SingleTaskGP to use as prior
+            transfer_mode: "mean" extracts posterior mean as prior mean,
+                          "kernel" uses prior's covariance
+            kernel_factory: Kernel factory for new covariance (required for mean mode,
```

Comment:
Cant we use default kernel factory (similar as single task GP) if it is none? I think I also followed that design in initial code.